### PR TITLE
test: context: Refine descriptions for some test cases

### PR DESCRIPTION
**Pull request by mengxianglinx**
Signed-off-by: Meng xianglin <xianglinx.meng@intel.com>

### CODE REVIEW

File: tests/kernel/context/src/main.c [hunk 1]
```diff
@@ -135,6 +135,84 @@ static struct k_thread thread_data3;
...
+ * - HAS_POWERSAVE_INSTRUCTION is set
+ *
+ * Input Specifications:
+ * - N/A
+ *
+ * Test Procedure:
+ * -# Record system time befor cpu enter idle state
+ * -# Enter cpu idle state by k_cpu_idle()
+ * -# Record system time after cpu idle state is intterupted
+ * -# Compare the two sytem time values.
```
Comment (enjiamai) [inline tests/kernel/context/src/main.c:161]: typo, "system"

File: tests/kernel/context/src/main.c [hunk 2]
```diff
@@ -135,6 +135,84 @@ static struct k_thread thread_data3;
...
+ * Prerequisite Condition:
+ * - HAS_POWERSAVE_INSTRUCTION is set
+ *
+ * Input Specifications:
+ * - N/A
+ *
+ * Test Procedure:
+ * -# Record system time befor cpu enter idle state
+ * -# Enter cpu idle state by k_cpu_idle()
+ * -# Record system time after cpu idle state is intterupted
```
Comment (enjiamai) [inline tests/kernel/context/src/main.c:160]: typo, "interrupted"

File: tests/kernel/context/src/main.c [hunk 3]
```diff
@@ -135,6 +135,84 @@ static struct k_thread thread_data3;
 
 static ISR_INFO isr_info;
 
+/**
+ * @brief Tes cpu idle function
```
Comment (enjiamai) [inline tests/kernel/context/src/main.c:139]: typo, "Test"

File: tests/kernel/context/src/main.c [hunk 4]
```diff
@@ -135,6 +135,84 @@ static struct k_thread thread_data3;
...
+ * - Interface testing
+ *
+ * Prerequisite Condition:
+ * - HAS_POWERSAVE_INSTRUCTION is set
+ *
+ * Input Specifications:
+ * - N/A
+ *
+ * Test Procedure:
+ * -# Record system time befor cpu enter idle state
```
Comment (enjiamai) [inline tests/kernel/context/src/main.c:158]: typo, "before"

File: tests/kernel/context/src/main.c [hunk 5]
```diff
@@ -135,6 +135,84 @@ static struct k_thread thread_data3;
...
+ * @ingroup kernel_context_tests
+ */
+static void test_kernel_cpu_idle(void);
+
+/**
+ * @brief Tes cpu idle function
+ *
+ * @details
+ * Test Objectve:
+ * - The kenel architecture provide a idle function to be run when the system
```
Comment (enjiamai) [inline tests/kernel/context/src/main.c:182]: typo, "kernel"

File: tests/kernel/context/src/main.c [hunk 6]
```diff
@@ -135,6 +135,84 @@ static struct k_thread thread_data3;
...
+ * - HAS_POWERSAVE_INSTRUCTION is set
+ *
+ * Input Specifications:
+ * - N/A
+ *
+ * Test Procedure:
+ * -# Record system time befor cpu enter idle state
+ * -# Enter cpu idle state by k_cpu_atomic_idle()
+ * -# Record system time after cpu idle state is intterupted
+ * -# Compare the two sytem time values.
```
Comment (enjiamai) [inline tests/kernel/context/src/main.c:200]: typo "interrupted" and "system"

File: tests/kernel/context/src/main.c [hunk 7]
```diff
@@ -472,16 +550,43 @@ static void test_kernel_timer_interrupts(void)
 }
 
 /**
+ * @brief Test some context routines
  *
- * @brief Test some context routines from a preemptible thread
+ * @details
+ * Test Objectve:
+ * - Thread contex handles derived from context switches must be able to be
```
Comment (enjiamai) [inline tests/kernel/context/src/main.c:557]: type, should be "context"?

File: tests/kernel/common/src/clock.c [hunk 8]
```diff
@@ -85,8 +85,43 @@ void test_clock_uptime(void)
 /**
  * @brief Test clock cycle functionality
  *
+ * @details
+ * Test Objectve:
+ * - The kenel architecture provide a 32bit monotonically increasing
```
Comment (enjiamai) [inline tests/kernel/common/src/clock.c:90]: typo, "kernel"

File: tests/kernel/context/src/main.c [hunk 9]
```diff
@@ -135,6 +135,84 @@ static struct k_thread thread_data3;
...
+ * Assumptions and Constraints
+ * - N/A
+ *
+ * @see k_cpu_idle()
+ * @ingroup kernel_context_tests
+ */
+static void test_kernel_cpu_idle(void);
+
+/**
+ * @brief Tes cpu idle function
```
Comment (enjiamai) [inline tests/kernel/context/src/main.c:178]: typo, "test"

**Review by enjiamai — CHANGES_REQUESTED** [review]
Please correct the typos in the descriptions.

File: tests/kernel/context/src/main.c [hunk 10]
```diff
@@ -135,6 +135,84 @@ static struct k_thread thread_data3;
 
 static ISR_INFO isr_info;
 
+/**
+ * @brief Test cpu idle function
+ *
+ * @details
+ * Test Objectve:
+ * - The kenel architecture provide an idle function to be run when the system
```
Comment (enjiamai) [inline tests/kernel/context/src/main.c:143]: typo "kernel"

**Review by enjiamai — APPROVED** [review]

File: tests/kernel/common/src/clock.c [hunk 11]
```diff
@@ -85,8 +85,43 @@ void test_clock_uptime(void)
...
+ *
+ * Pass/Fail criteria:
+ * - Success if cycles increase monotonically, failure otherwise.
+ *
+ * Test Procedure:
+ * -# At mili-second boundary, get cycles repeatedly by k_cycle_get_32()
+ *  till cycles increased
+ * -# At mili-second boundary, get cycles repeatedly by k_uptime_get_32()
+ *  till cycles increased
+ * -# Cross check cycles get by k_cycle_get_32() and k_uptime_get_32(),
```
Comment (LeiW000) [inline tests/kernel/common/src/clock.c:116]: It seems to me that "get" should be "gotten".

File: tests/kernel/common/src/clock.c [hunk 12]
```diff
@@ -85,8 +85,43 @@ void test_clock_uptime(void)
...
+ *  the delta cycle should be greater 1 milli-second.
```
Comment (LeiW000) [inline tests/kernel/common/src/clock.c:117]: Missing "than"?

File: tests/kernel/context/src/main.c [hunk 13]
```diff
@@ -135,6 +135,84 @@ static struct k_thread thread_data3;
...
+ * - Interface testing
+ *
+ * Prerequisite Condition:
+ * - HAS_POWERSAVE_INSTRUCTION is set
+ *
+ * Input Specifications:
+ * - N/A
+ *
+ * Test Procedure:
+ * -# Record system time befor cpu enter idle state
```
Comment (LeiW000) [inline tests/kernel/context/src/main.c:197]: "enter" should be "enters"

**Review by LeiW000 — CHANGES_REQUESTED** [review]

**Review by LeiW000 — APPROVED** [review]
I think it can be merged now.

**Pull request merged**